From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: sub-kek <subbkek@gmail.com>
Date: Mon, 30 Dec 2024 13:20:53 +0300
Subject: [PATCH] Stonecutter damage


diff --git a/src/main/java/org/bukkit/event/entity/EntityDamageEvent.java b/src/main/java/org/bukkit/event/entity/EntityDamageEvent.java
index d1a5424ff3b289f1c82449ef6d88eb52665df41b..da4f6f4494bfc4601096388c6739c9e894d000f0 100644
--- a/src/main/java/org/bukkit/event/entity/EntityDamageEvent.java
+++ b/src/main/java/org/bukkit/event/entity/EntityDamageEvent.java
@@ -498,6 +498,7 @@ public class EntityDamageEvent extends EntityEvent implements Cancellable {
          * <p>
          * Damage: 10
          */
-        SONIC_BOOM;
+        SONIC_BOOM,
+        STONECUTTER; // Etheria
     }
 }
